src/annoylib.h

```diff
@@ -164,8 +148,14 @@ struct Angular {
       return random->flip();
   }
   static inline void create_split(const vector<node*>& nodes, int f, Randomness<T>* random, node* n) {
+    // Sample two random points from the set of nodes
+    // Calculate the hyperplane equidistant from them
+    int i = random->index(nodes.size());
+    int j = random->index(nodes.size());
```

Comment:
Ok I get what you are saying. Makes sense.

I think you still want to pick the first vector not uniformly randomly but somehow skewed to the distribution of the data set. If you have a very high dimensional dataset that is basically some low dimensional hyperplane H plus a bunch of noise then the new 2-point sampling algorithm will generate splits approximately perpendicular to H which seems beneficial